classes/class-p4-master-site.php

```diff
@@ -868,4 +799,80 @@ public function filter_youtube_oembed_nocookie( $cache, $url ) {
 		return $cache;
 	}
 
+	/**
+	 * Add custom media metadata fields.
+	 *
+	 * @param array    $form_fields An array of fields included in the attachment form.
+	 * @param \WP_Post $post The attachment record in the database.
+	 *
+	 * @return array Final array of form fields to use.
+	 */
+	public function add_image_attachment_fields_to_edit( $form_fields, $post ) {
+
+		// Add a Credit field.
+		$form_fields['credit_text'] = [
+			'label' => __( 'Credit', 'planet4-master-theme-backend' ),
+			'input' => 'text', // this is default if "input" is omitted.
+			'value' => get_post_meta( $post->ID, '_credit_text', true ),
+			'helps' => __( 'The owner of the image.', 'planet4-master-theme-backend' ),
+		];
+
+		return $form_fields;
+	}
+
+	/**
+	 * Save custom media metadata fields
+	 *
+	 * @param \WP_Post $post        The $post data for the attachment.
+	 * @param array    $attachment  The $attachment part of the form $_POST ($_POST[attachments][postID]).
+	 *
+	 * @return \WP_Post $post
+	 */
+	public function add_image_attachment_fields_to_save( $post, $attachment ) {
+		if ( isset( $attachment['credit_text'] ) ) {
+			update_post_meta( $post['ID'], '_credit_text', $attachment['credit_text'] );
+		}
+
+		return $post;
+	}
+
+	/**
+	 * Override the Gutenberg core/image block render method output, To add credit field in it's caption text.
+	 *
+	 * @param array  $attributes    Attributes of the Gutenberg core/image block.
+	 * @param string $content The image element HTML.
+	 *
+	 * @return string HTML content of image element with credit field in caption.
+	 */
+	public function p4_core_image_block_render( $attributes, $content ) {
+		$image_id = trim( str_replace( 'attachment_', '', $attributes['id'] ) );
+		$meta     = get_post_meta( $image_id );
+
+		if ( isset( $meta['_credit_text'] ) && ! empty( $meta['_credit_text'][0] ) ) {
+			$image_credit = ' ' . $meta['_credit_text'][0];
+			if ( false === strpos( $meta['_credit_text'][0], '©' ) ) {
+				$image_credit = ' ©' . $image_credit;
+			}
+
+			if ( $image_credit ) {
```

Comment:
This condition will always be true as `$image_credit` comes from `$meta['_credit_text'][0]` which is already guaranteed from the check above to be non-empty. So it can just be removed here.